File: gemini_cli.py
#!/usr/bin/env python3
"""
Gemini CLI - Simple command-line interface for Google's Gemini AI
Usage: python gemini_cli.py "your question here"
"""

import sys
import os
import google.generativeai as genai
from google.generativeai.types import HarmCategory, HarmBlockThreshold

def setup_gemini():
    """Setup Gemini with API key from config.json or environment"""
    # Try to get API key from config.json first
    try:
        import json
        with open('config.json', 'r') as f:
            config = json.load(f)
            api_key = config.get('api_key')
            if api_key and api_key != 'YOUR_GEMINI_API_KEY_HERE':
                genai.configure(api_key=api_key)
                return genai.GenerativeModel('gemini-1.5-flash')
    except:
        pass
    
    # Fallback to environment variable
    api_key = os.getenv('GEMINI_API_KEY')
    if api_key:
        genai.configure(api_key=api_key)
        return genai.GenerativeModel('gemini-1.5-flash')
    
    # If no API key found
    print("❌ No Gemini API key found!")
    print("Please either:")
    print("1. Set GEMINI_API_KEY environment variable")
    print("2. Update config.json with your API key")
    print("3. Get your API key from: https://makersuite.google.com/app/apikey")
    return None

def chat_with_gemini(prompt, model):
    """Send prompt to Gemini and return response"""
    try:
        # Configure safety settings to be less restrictive
        safety_settings = {
            HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
            HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
            HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
            HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
        }
        
        response = model.generate_content(
            prompt,
            safety_settings=safety_settings,
            generation_config={
                "temperature": 0.7,
                "max_output_tokens": 1000,
            }
        )
        return response.text
    except Exception as e:
        return f"❌ Error: {str(e)}"

def main():
    if len(sys.argv) < 2:
        print("🤖 Gemini CLI")
        print("Usage: python gemini_cli.py 'your question here'")
        print("Example: python gemini_cli.py 'Explain quantum computing'")
        return
    
    prompt = ' '.join(sys.argv[1:])
    
    print("🤖 Initializing Gemini...")
    model = setup_gemini()
    
    if not model:
        return
    
    print(f"💭 Asking: {prompt}")
    print("🤖 Gemini is thinking...")
    
    response = chat_with_gemini(prompt, model)
    print(f"\n✨ Response:\n{response}")

if __name__ == "__main__":
    main()

File: gemini_cli.js
#!/usr/bin/env node
/**
 * Gemini CLI - Node.js version
 * Usage: node gemini_cli.js "your question here"
 */

import { GoogleGenerativeAI } from '@google/generative-ai';
import fs from 'fs';

function setupGemini() {
    // Try to get API key from config.json first
    try {
        const config = JSON.parse(fs.readFileSync('config.json', 'utf8'));
        const apiKey = config.api_key;
        if (apiKey && apiKey !== 'YOUR_GEMINI_API_KEY_HERE') {
            return new GoogleGenerativeAI(apiKey);
        }
    } catch (error) {
        // Fallback to environment variable
    }
    
    // Fallback to environment variable
    const apiKey = process.env.GEMINI_API_KEY;
    if (apiKey) {
        return new GoogleGenerativeAI(apiKey);
    }
    
    console.log("❌ No Gemini API key found!");
    console.log("Please either:");
    console.log("1. Set GEMINI_API_KEY environment variable");
    console.log("2. Update config.json with your API key");
    console.log("3. Get your API key from: https://makersuite.google.com/app/apikey");
    return null;
}

async function chatWithGemini(prompt, genAI) {
    try {
        const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
        
        const result = await model.generateContent(prompt, {
            safetySettings: [
                {
                    category: "HARM_CATEGORY_HATE_SPEECH",
                    threshold: "BLOCK_MEDIUM_AND_ABOVE"
                },
                {
                    category: "HARM_CATEGORY_DANGEROUS_CONTENT", 
                    threshold: "BLOCK_MEDIUM_AND_ABOVE"
                },
                {
                    category: "HARM_CATEGORY_SEXUALLY_EXPLICIT",
                    threshold: "BLOCK_MEDIUM_AND_ABOVE"
                },
                {
                    category: "HARM_CATEGORY_HARASSMENT",
                    threshold: "BLOCK_MEDIUM_AND_ABOVE"
                }
            ],
            generationConfig: {
                temperature: 0.7,
                maxOutputTokens: 1000,
            }
        });
        
        return result.response.text();
    } catch (error) {
        return `❌ Error: ${error.message}`;
    }
}

async function main() {
    if (process.argv.length < 3) {
        console.log("🤖 Gemini CLI (Node.js)");
        console.log("Usage: node gemini_cli.js 'your question here'");
        console.log("Example: node gemini_cli.js 'Explain quantum computing'");
        return;
    }
    
    const prompt = process.argv.slice(2).join(' ');
    
    console.log("🤖 Initializing Gemini...");
    const genAI = setupGemini();
    
    if (!genAI) {
        return;
    }
    
    console.log(`💭 Asking: ${prompt}`);
    console.log("🤖 Gemini is thinking...");
    
    const response = await chatWithGemini(prompt, genAI);
    console.log(`\n✨ Response:\n${response}`);
}

main().catch(console.error);
